pkg/controllers/clusterresourceplacement/controller.go

```diff
@@ -179,6 +186,37 @@ func (r *Reconciler) getOrCreateClusterPolicySnapshot(ctx context.Context, crp *
 	return latestPolicySnapshot, nil
 }
 
+func (r *Reconciler) deleteRedundantSchedulingPolicySnapshots(ctx context.Context, crp *fleetv1beta1.ClusterResourcePlacement, sortedList *fleetv1beta1.ClusterSchedulingPolicySnapshotList) error {
+	var err error
+	if sortedList == nil {
+		sortedList, err = r.listSortedClusterSchedulingPolicySnapshots(ctx, crp)
+		if err != nil {
+			return err
+		}
+	}
+	crpKObj := klog.KObj(crp)
+	// respect the revisionHistoryLimit field
+	revisionLimit := fleetv1beta1.RevisionHistoryLimitDefaultValue
+	if crp.Spec.RevisionHistoryLimit != nil {
+		revisionLimit = *crp.Spec.RevisionHistoryLimit
+		if revisionLimit <= 0 {
```

Comment:
This check should be done on the schema end; and it would be better to use the default value if the setting is invalid rather than throwing an error here IMO.